bpo-41842: Add a unregister function in _codecs module (#22360)
python/cpython: Doc/library/codecs.rst

   :class:`CodecInfo` object. In case a search function cannot find
   a given encoding, it should return ``None``.

.. function:: unregister(search_function)

   Unregister a codec search function from the codecs registry.
   If the search function is not registered, do nothing.

>    Clear the registry's cache if the codec search function is removed.

Reviewer: We can avoid the repetition "if the function is removed" like that:

```suggestion
   Unregister a codec search function and clear the registry's cache.
   If the search function is not registered, do nothing.
```

I don't think that it's useful to specify "from the codecs registry": the codecs.register() documentation doesn't mention the registry.

Author: ok, forgive my poor English.

   .. note::

      Search function registration is reversible since Python 3.10.

>       You can use `codecs.unregister()` to unregister the search function.

Reviewer: I suggest to remove the note.

Author: done

.. function:: unregister(search_function)

>    Unregister a codec search function from the codecs registry. If
>    the search function haven't registered, this function dose nothing.
> 

Reviewer: ```suggestion
   Unregister a codec search function from the codecs registry.
   
   If the search function is not registered, do nothing.
```

Author: done
